v2_test.go

```diff
@@ -483,17 +484,6 @@ var tlvFormatTests = []struct {
 			rawTLVs:           make([]byte, 1<<16),
 		},
 	},
-	{
-		desc: "local unspec",
```

Comment:
Why is this test deleted? IIUC only addresses are ignored and attributes like TLVs and Transport Protocol.

Reply:
@pires The reason I delete this test case is that in our unit test cases (based on packets received from AWSLB), if local connection is presented, the rest of the headers and TLVs are ignored. That is why in v2.go, we added a check to see if the header is local, and if so, the function just returns the header without coping the remaining headers and the TLV data. 

I reviewed the latest proxy protocol spec https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt but couldn't find any information indicating whether the receiver should or should not keep the TLV data if the connection is local. We assume that the remaining headers and TLVs are ignored in this case. cc @igor-kupczynski to get some feedback.

Reply:
The spec is clear that when LOCAL is presented, the receiver is to ignore the address information, nothing else. Implicitly, you can still pass LOCAL+<INET FAMILY>+TLVs so let's keep that until the standard makes it clear. I also don't see any problem in keeping this as this information is only processed if it's there.

Reply:
@pires Thank you for your feedback. Just want to provide a little bit more context. The reason we ignore the TLVs if it is local connection because we captured the following packets from AWS and Azure:

AWS:
```
awsLocalPP := []byte{
		'\x0d', '\x0a', '\x0d', '\x0a', '\x00', '\x0d', '\x0a', '\x51', '\x55', '\x49', '\x54', '\x0a', //v2 sig
		'\x20',         // command LOCAL
		'\x11',         // Address Family TCP over IPv4
		'\x00', '\x54', // length 84 bytes
		'\x03', '\x00', '\x04', '\xbc', '\xe5', '\x95', '\x1f', // This is actually a CRC type 3 TLV
		'\x04', '\x00', '\x4a', // This is a NOOP TLV which covers the rest of the nul bytes in the header
		'\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00',
		'\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00',
		'\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00',
		'\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00',
		'\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00',
		'\x00', '\x00', '\x00', '\x00',
	}
```
Azure:
```
azureLocalHealthPP := []byte{
		'\x0d', '\x0a', '\x0d', '\x0a', '\x00', '\x0d', '\x0a', '\x51', '\x55', '\x49', '\x54', '\x0a', // v2 sig
		'\x20',         // command LOCAL
		'\x00',         // address family ??
		'\x00', '\x0c', // length
		'\x00', '\x00', '\x00', '\x00',
		'\x00', '\x00', '\x00', '\x00',
		'\x00', '\x00',
		'\x00', '\x00',
	}
```
It seems that AWS sets the TLVs to all zeros if it is local connection and Azure drops the TLVs. That is why in our implementation, we ignore the TLV data. 

I've talked to @igor-kupczynski and Franek (another contributor), we also think that it should be ok to pass TLVs if the connection is local. I am going to run more tests to confirm.

I will provide an update once I have more test results and if everything is ok, we could probably close this PR.

Reply:
I'm waiting on the tests to be added again, just to be sure we're on the same page.

Reply:
@pires Thank you for following with this PR. Currently I am running e2e integration tests on aws, gcp and azure. I don't have the test results yet and will provide an update once my testing is completed.